File: 530-Minimum-Absolute-Difference-in-BST/solution.py
class Solution:
    def getMinimumDifference(self, root):
        stack = []
        curr = root
        prev_val = None
        min_diff = float('inf')
        
        while stack or curr:
            while curr:
                stack.append(curr)
                curr = curr.left
            
            curr = stack.pop()
            
            if prev_val is not None:
                min_diff = min(min_diff, abs(curr.val - prev_val))
                
            prev_val = curr.val
            curr = curr.right
        
        return min_diff

File: 530-Minimum-Absolute-Difference-in-BST/solution.java
class Solution {
    public int getMinimumDifference(TreeNode root) {
        Stack<TreeNode> stack = new Stack<>();
        TreeNode curr = root;
        Integer prevVal = null;
        int minDiff = Integer.MAX_VALUE;
        
        while (!stack.isEmpty() || curr != null) {
            while (curr != null) {
                stack.push(curr);
                curr = curr.left;
            }
            
            curr = stack.pop();
            
            if (prevVal != null) {
                minDiff = Math.min(minDiff, Math.abs(curr.val - prevVal));
            }
            
            prevVal = curr.val;
            curr = curr.right;
        }
        
        return minDiff;
    }
}